src/lib/Components/form-group-component/form-group-component.ts

```diff
@@ -42,11 +43,20 @@ export class FormGroupComponent {
     );
   }
 
+  public errorMessages: ErrorMessage[];
+
   constructor(
     private elRef: ElementRef,
     private errorMessageService: ErrorMessageService
   ) {}
 
+  ngOnInit() {
+    this.errorMessages = [
+      ...this.errorMessageService.errorMessages,
+      ...this.customErrorMessages
+    ].reverse();
```

Comment:
What's the reasoning for switching the order and then using `reverse()` here?

Reply:
Because of the order provider by the application injector `['App_ErrMsgs', 'Child_ErrMsgs', 'Child_Child_ErrMsgs']` so we add the component level messages last as they would have "highest" priority. Then since find works left -> right we need to reverse it so maintain priority.

Reply:
Ah, gotcha.